From 5fd53f5bdeb648ca8f560aff8e33570fb68db39b Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Mon, 23 Jul 2018 12:35:37 -0700
Subject: [PATCH] TRITON-622 'stdin.write' in promptYesNo func in common.js
 breaks on node v10 Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
 Approved by: Todd Whiteman <todd.whiteman@joyent.com>

---
 src/img/CHANGES.md    | 4 ++++
 src/img/lib/common.js | 2 +-
 src/img/package.json  | 2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index b3260374..31fd9b1e 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -5,6 +5,10 @@ Known issues:
 - Docker image imports are experimental. Docker image import also only supports
   Docker Registry v2.
 
+## 3.9.2
+
+- TRITON-622 'stdin.write' in promptYesNo func in common.js breaks on node v10
+
 ## 3.9.1
 
 - OS-5979 ensure imgadm uses the provided req_id
diff --git a/src/img/lib/common.js b/src/img/lib/common.js
index 59e54bd0..7479cabd 100644
--- a/src/img/lib/common.js
+++ b/src/img/lib/common.js
@@ -275,9 +275,9 @@ function promptYesNo(opts_, cb) {
     stdin.on('data', onData);
 
     function postInput() {
+        stdout.write('\n');
         stdin.setRawMode(false);
         stdin.pause();
-        stdin.write('\n');
         stdin.removeListener('data', onData);
     }
 
diff --git a/src/img/package.json b/src/img/package.json
index 2b968dfe..8112fc9a 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.9.1",
+  "version": "3.9.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

